From 0d34a3ca5806e4f12ea4c26b9628732c01951118 Mon Sep 17 00:00:00 2001
From: Mitch Curtis <mitch.curtis@qt.io>
Date: Thu, 28 Aug 2025 10:40:15 +0800
Subject: [PATCH 232/239] tst_objectcount: tidy up code

- Move exclusions common to all tests into a variable that we can reuse
- Keep one entry per line to keep git blame useful
- Sort entries alphabetically

Change-Id: I28f2816e3431bb7ca22f6d8a019354a4282fded1
Reviewed-by: Oliver Eftevaag <oliver.eftevaag@qt.io>
(cherry picked from commit 1e07120953ea4454d27817cad0602bf0c3df09a8)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../objectcount/tst_objectcount.cpp           | 71 +++++++++++--------
 1 file changed, 43 insertions(+), 28 deletions(-)

diff --git a/tests/benchmarks/quickcontrols/objectcount/tst_objectcount.cpp b/tests/benchmarks/quickcontrols/objectcount/tst_objectcount.cpp
index d4c0a610e7..8dbd1f1d64 100644
--- a/tests/benchmarks/quickcontrols/objectcount/tst_objectcount.cpp
+++ b/tests/benchmarks/quickcontrols/objectcount/tst_objectcount.cpp
@@ -62,38 +62,53 @@ void tst_ObjectCount::cleanup()
 
 static void initTestRows(QQmlEngine *engine)
 {
-    // Calendar is excluded because it's a singleton and can't be created.
-    // TreeViewDelegate is excluded since it's a delegate that can only be created by TreeView.
-    // TableViewDelegate is excluded since it's a delegate that can only be created by TableView.
-    // Horizontal/VerticalHeaderViewDelegate is excluded since it's a delegate
-    // that can only be created by Horizontal/VerticalHeaderView.
+    const QStringList commonExclusions = {
+        // Singleton; can't be created.
+        "Calendar",
+        // Can only be created by HorizontalHeaderView.
+        "HorizontalHeaderViewDelegate",
+        // Can only be created by TreeView.
+        "TreeViewDelegate",
+        // Can only be created by TableView.
+        "TableViewDelegate",
+        // Can only be created by VerticalHeaderView.
+        "VerticalHeaderViewDelegate"
+    };
+
     addTestRowForEachControl(engine, QQC2_IMPORT_PATH, "basic", "QtQuick/Controls/Basic",
-                             QStringList() << "Calendar"
-                                           << "TreeViewDelegate"
-                                           << "TableViewDelegate"
-                                           << "HorizontalHeaderViewDelegate"
-                                           << "VerticalHeaderViewDelegate");
-    addTestRowForEachControl(
-            engine, QQC2_IMPORT_PATH, "fusion", "QtQuick/Controls/Fusion",
-            QStringList() << "ButtonPanel" << "CheckIndicator" << "RadioIndicator" << "SliderGroove"
-                          << "SliderHandle" << "SwitchIndicator" << "TreeViewDelegate"
-                          << "HorizontalHeaderViewDelegate"
-                          << "VerticalHeaderViewDelegate");
+        commonExclusions);
+
+    addTestRowForEachControl(engine, QQC2_IMPORT_PATH, "fusion", "QtQuick/Controls/Fusion",
+        QStringList()
+            << commonExclusions
+            << "ButtonPanel"
+            << "CheckIndicator"
+            << "RadioIndicator"
+            << "SliderGroove"
+            << "SliderHandle"
+            << "SwitchIndicator");
+
     addTestRowForEachControl(engine, QQC2_IMPORT_PATH, "imagine", "QtQuick/Controls/Imagine",
-                             QStringList() << "HorizontalHeaderViewDelegate"
-                                           << "VerticalHeaderViewDelegate");
+        commonExclusions);
+
     addTestRowForEachControl(engine, QQC2_IMPORT_PATH, "material", "QtQuick/Controls/Material",
-                             QStringList() << "Ripple" << "SliderHandle" << "CheckIndicator"
-                                           << "RadioIndicator"
-                                           << "SwitchIndicator" << "BoxShadow" << "ElevationEffect"
-                                           << "CursorDelegate" << "TreeViewDelegate"
-                                           << "HorizontalHeaderViewDelegate"
-                                           << "VerticalHeaderViewDelegate");
+        QStringList()
+           << commonExclusions
+           << "BoxShadow"
+           << "CheckIndicator"
+           << "CursorDelegate"
+           << "ElevationEffect"
+           << "RadioIndicator"
+           << "Ripple"
+           << "SliderHandle"
+           << "SwitchIndicator");
+
     addTestRowForEachControl(engine, QQC2_IMPORT_PATH, "universal", "QtQuick/Controls/Universal",
-                             QStringList()
-                                     << "CheckIndicator" << "RadioIndicator" << "SwitchIndicator"
-                                     << "HorizontalHeaderViewDelegate"
-                                     << "VerticalHeaderViewDelegate");
+        QStringList()
+            << commonExclusions
+            << "CheckIndicator"
+            << "RadioIndicator"
+            << "SwitchIndicator");
 }
 
 template <typename T>
-- 
2.51.2

